### PR TITLE
Database

### PR DESCRIPTION
**Pull request by theBwoyProgrammer**
# Query and Update
- Altered table to add species column
- Added more animals

## Altered
altered

## Inserted
inserted

## Updated and Rollback
updated and rollback

## Update species
updated species

## Delete and Rollback
delete rollback

## Deleted and savepoint set
deleting savepoint weight

## Rollback and Commit
rollback commit weight

## Queries to Questions
qns

## Neutered, not neutered and min, max weight of each type of animal
image

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -9,3 +9,7 @@ CREATE TABLE animals (
     weight_kg DECIMAL NOT NULL,
     PRIMARY KEY(id),
 );
```
Comment (HAMMAS-SALEEM) [inline schema.sql]: - [x] Please remove comma at the end of line 10 because it is causing an syntax error :fearful:

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,67 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
...
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT MAX(escape_attempts) FROM animals WHERE neutered = true OR false;
+
```
Comment (HAMMAS-SALEEM) [inline queries.sql]: - [x] The requirement is to get the animal who escaped the most from both types neutered and not neutered. Kindy query that animal who escaped the most from both types. You can use group by to get most escaped animal from both types :+1:

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,67 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
...
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg), MAX(weight_kg) FROM animals;
+
```
Comment (HAMMAS-SALEEM) [inline queries.sql]: - [x] Please give maximum and minimum weight animal from each type (pokemon and digimon). You can use group by to get maximum and minimum weight from both types. :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:

almost-there


## Hey @HatimDev-HE :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Following GitFlow
:heavy_check_mark: Descriptive PR Summary
:heavy_check_mark: Following Some Requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (theBwoyProgrammer) [reply schema.sql]: done

File: queries.sql (hunk 3 above)
Comment (Davidosky007) [reply queries.sql]: * As stated by the previous reviewer your query needs to select ` the minimum and maximum weight of each type of animal?` which according to your transaction query below you have grouped your animals into two types `pokemon and digimon` kindly write your queries to target these two types of animals selecting their min weight and max weight grouped by species. HINT 💭 Your query can look like this `
SELECT TYPE OF ANIMAL, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY TYPE OF ANIMAL;
`
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @HatimDev-HE 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Good looking readme file 👍 
* Well written PR description 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (theBwoyProgrammer) [reply queries.sql]: done

File: queries.sql (hunk 3 above)
Comment (theBwoyProgrammer) [reply queries.sql]: done

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @HatimDev-HE,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**